src/context/auth/Context.jsx

```diff
@@ -0,0 +1,86 @@
+import {
+  createContext,
+  useContext,
+  useReducer,
+  useState,
+  useEffect,
+} from 'react';
+import { authReducer } from './Reducer';
+import { ACTION_TYPES, axios, requests } from '../../utils';
+import { useNavigate } from 'react-router-dom';
+
+const AuthContext = createContext();
+
+const useAuth = () => useContext(AuthContext);
+
+const AuthProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(authReducer, {
+    token: localStorage.getItem('token'),
+    user: null,
+  });
+
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState();
+
+  const navigate = useNavigate();
+
+  const handleSubmit = async (data, path, route) => {
+    setLoading(true);
+    try {
+      const res = await axios.post(route, data);
+      if (res.status === 200 || res.status === 201) {
+        authDispatch({ type: ACTION_TYPES.LOGIN, payload: res.data });
+        localStorage.setItem('token', res.data.token);
+      }
+      navigate(path, { replace: true });
+    } catch (error) {
+      console.log(error);
```

Comment:
better to use a throw Error instead

Reply:
Removed `console.log()` here, Not using throw Error because handling error with `setError(error)` to show error msg to the user.